xarray/backends/api.py

```diff
@@ -565,8 +565,11 @@ def to_netcdf(dataset, path=None, mode='w', format=None, group=None,
     sync = writer is None
 
     store = store_cls(path, mode, format, group, writer)
+    # Copy dataset encoding to datastore
+    store.encoding = dataset.encoding
```

Comment:
Do we ever actually use this `encoding` state on the datastore? If not, let's not bother setting it. I think everything necessary ends up being passed on via `set_variables`.

Note that as bunch as possible, I've tried to make `DataStore` itself stateless, only storing state in the file-like object it points to.